### PR TITLE
[WIP] MAID-2688 chore/connectivity: enable reconnect for Peruse app and update state on network status

### PR DESCRIPTION
**Pull request by hunterlester**
~Why WIP? Because this task is blocking while figure out how to test the implementation without running into https://maidsafe.atlassian.net/browse/MAID-2639.~

### CODE REVIEW

File: app/extensions/safe/network/authenticator-comms.js [hunk 1]
```diff
@@ -96,7 +97,6 @@ export const initAnon = async ( passedStore ) =>
 
     try
     {
-        // does it matter if we override?
         const networkStateCb = (state) => {
```
Comment (joshuef) [inline app/extensions/safe/network/authenticator-comms.js]: @hunterlester you mean this line is not being triggered, right?

**Comment by hunterlester** [comment]
Requires local link of https://github.com/maidsafe/safe_app_nodejs/pull/251 to fire network callback when unregistered connection event occurs.
Network callback fires upon connection and state dispatch is happening but I'm unable to observe updated state: https://github.com/maidsafe/safe_browser/pull/262/files#diff-c1373fedfea725334ffc2413bc67f357R88

Also for test, error is thrown when attempting to import function, https://github.com/maidsafe/safe_browser/pull/262/files#diff-a003f29f7e2f9aeecfe6e3fbb39e3d2eR8, for same reason as before where other operations in the module are initialised, one or two of which rely on a main process.

File: app/extensions/safe/network/authenticator-comms.js [hunk 2]
```diff
@@ -76,6 +79,42 @@ export const handleSafeAuthAuthentication = ( uriOrReqObject, type ) =>
...
 
+export const onNetworkStateChange = (store) => (state) =>
+{
+  logger.info('net callback state: ', state);
+  const previousState = store.getState().peruseApp.networkStatus;
+  logger.info('previousState: ', previousState);
+  store.dispatch( setNetworkStatus( state ) );
+  // TODO/BOOKMARK: Expected is an updated store state but not happening
+  logger.info('store state: ', store.getState());
+  const newState = store.getState().peruseApp.networkStatus;
```
Comment (joshuef) [inline app/extensions/safe/network/authenticator-comms.js]: @hunterlester the `dispatch` calls happen async, so here, when you're pinging for `getState` again, it hasn't necessarily updated the store.

But you don't need to ping the store for this anyway, the passed `state` to this function is your newState.

File: app/extensions/safe/network/authenticator-comms.js [hunk 3]
```diff
@@ -76,6 +79,42 @@ export const handleSafeAuthAuthentication = ( uriOrReqObject, type ) =>
...
+      store.dispatch( addNotification(
+        {
+          text: `Network state: ${newState}. Reconnecting...`,
+          type: 'error',
+          onDismiss: clearNotification
+        } 
+      ));
+    }
+  }
+  while (isDisconnected) {
```
Comment (joshuef) [inline app/extensions/safe/network/authenticator-comms.js]: I dont think this while is what's needed here (`isDisconnected` wont change in this function call...)

if you set up a func that will check the state after the timeout, and recall itself if needed, that should do the trick:
```js

  const attemptReconnect = ( store ) =>
  {
      // logger.info(' in while disconnected')
      setTimeout(() => {
          logger.info('Attempting reconnect...');
          peruseAppObj.reconnect();

          if( store.getState.peruseApp.networkStatus === SAFE.NETWORK_STATE.DISCONNECTED )
          {
              attemptReconnect(store );
          }
      }, 5000);

  }
```

**Comment by hunterlester** [comment]
@joshuef Reconnect notification and operation running nicely upon `Disconnect` event.
If test is attempted, it must be linked with https://github.com/maidsafe/safe_app_nodejs/pull/252 in order to prevent false `Connected` when `app_reconnect` is called.

Almost there with the state update test. Need a little help with mocking store, see comment: https://github.com/maidsafe/safe_browser/pull/262/files#diff-a003f29f7e2f9aeecfe6e3fbb39e3d2eR84
I attempted to utilise Redux's `createStore` function but it doesn't quite replicate the same behavior as in the app's `configureStore` which relies on main process.
Reading [Jest Mock Functions](https://facebook.github.io/jest/docs/en/mock-function-api.html), the docs are unclear to me.

**Comment by joshuef** [comment]
@hunterlester my missing comment was that I was still seeing the error post-reconnect when you'd been trying to load a page.

**Comment by joshuef** [comment]
@hunterlester CI tests are failing here.

**Comment by hunterlester** [comment]
Closing PR to iterate over CI tests on my own fork.

**Pull request closed without merging**